### ISSUE TITLE
Workshop length & start time

### ISSUE THREAD

**Issue by robertkowalski** [nodeschool/havana#5]
Hi!

I am preparing a small website for the nodeschool and I wonder how long our workshop should be. I would suggest that it should not take longer than 6 hours, as the mentors and mentees get tired.

I also don't know at which time we should start. When does the location open and how long?

**Comment by olemis** [comment]
+1 ... ideally three hours in the morning and three in the afternoon ? BTW , what will be the subjects ?

@elMor3no has the other answers .

**Comment by elMor3no** [comment]
This is an draft of the schedule....
    Actividades  
Hora  
09:00 – 09:50 Breakfast / Opening Session  
10:00 – 10:50 Keynote  
11:00 – 12:45 WorkShop / Sprints  
12:45 – 14:00 Lunch  
14:00 – 17:45 WorkShop / Sprints  WorkShop / Sprints  WorkShop / Sprints
17:45 – 18:30 Closing Session     

2016-01-29 10:16 GMT-05:00, Olemis Lang notifications@github.com:

**Comment by robertkowalski** [comment]
ah, cool! so we have a block of 2:25 hr and one block of 3:45

the attendees should have node.js installed on their machines, because our time is limited.

I think it makes sense to work on the "javascript basics" and "nodejs basics" workshoppers. you can see them here: http://nodeschool.io/es/#workshopper-list

here are more informations about workshopper: http://nodeschool.io/es/about.html

**Comment by olemis** [comment]
I would not take this for granted , not because people are lazy , but especially due to the fact that [internet access is problematic](https://github.com/nodeschool/organizers/wiki/Venue-Best-Practices#how-to-pick-the-right-venue) . The most practical approach I'd suggest consists in :
- create a (Virtualbox?) VM for the workshop with Vagrant (or alike)
- Ensure it is equipped with everything needed for the workshop .
- offer it for download (online and via a local WiFi network deployed at the venue)
- offer Vagrant packages for different platforms for download as well

@elMor3no is it possible to have a local Ubuntu repository (if possible with multiple versions) deployed at the venue ?

**Comment by elMor3no** [comment]
We can have a Ubuntu/Debian Repository.

Anyway is important if we have all the possible thing we will have off
line or in a local server with wifi

2016-01-29 17:10 GMT-05:00, Olemis Lang notifications@github.com:

**Comment by olemis** [comment]
I noticed that my previous message might be confusing .

@robertkowalski when I said "I would not take this for granted" this was related to "the attendees should have node.js installed on their machines" . I'm ok with the two subjects you proposed . I'd really like to see the functional programming workshop included ... but I guess there will be no much time to so ?

**Comment by olemis** [comment]
@robertkowalski besides, if you decide to do so (prepare the VM , etc ...) please let us know of the versions of Ubuntu/Debian/other OS, Virtualbox (or any other virtualization tool you choose), and Vagrant so that we all be in sync . Thanks in advance !

**Comment by robertkowalski** [comment]
@olemis we can also offer the functional programming workshop, attendees are able to pick from the three workshoppers based on their skills then. did you already take a look at the functional workshopper to get a feeling for the tasks?

@olemis @elMor3no:
- did you read https://github.com/nodeschool/organizers/wiki/Event-Mentor-Best-Practices ? have you completed the first two workshoppers?
- do we already have other people who are interested in mentoring and are from cuba, but not on github?

**Comment by olemis** [comment]
the functional workshop ? really ? @elMor3no is there enough time ?

**Comment by olemis** [comment]
about the workshoppers , consider it done by next week 

about potential mentors , we'll have a meeting on Feb 13th and I'm hoping others will join us .

## 

Regards

**Comment by robertkowalski** [comment]
@olemis I would not expect that we have enough time for people to finish all three during the event. but they could chose between the three based on their skillset. if we have advanced users they are probably bored by the JS fundamentals workshop. does that make sense?

**Comment by robertkowalski** [comment]
@olemis cool that is great to hear! :)

**Comment by elMor3no** [comment]
El 03/02/16 19:17, Olemis Lang escribió:

> the functional workshop ? really ? @elMor3no 
> https://github.com/elMor3no is there enough time ?
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/nodeschool/havana/issues/5#issuecomment-179546038.
> 
> How much time will take?

## 

Grupo de Usuarios de Tecnologías Libres - Cuba
http://gutl.jovenclub.cu/

**Comment by robertkowalski** [comment]
@elMor3no how much time an attendee needs to complete a workshopper strongly depends on the attendee's skill set and previous experience. many people don't finish one workshopper during a whole day and that is perfectly ok.

it is also depending on the skill set of the attendees which workshops we _can_ offer to them. a experienced JS developer will be bored by the JS fundamentals workshopper, but probably be interested in something like the functional programming workshopper. I would expect a mixed group of people (some with no experience at all and some experienced). we should not demand that everyone has to do the JS fundamentals workshop, when it would bore them.

a person who has never used javascript will probably be a bit overwhelmed in the beginning and need some more help to get started. i would not expect them to finish the two workshoppers and then start the functional workshopper right after it.

**Comment by elMor3no** [comment]
El 05/02/16 12:00, Robert Kowalski escribió:

> @elMor3no https://github.com/elMor3no how much time an attendee 
> needs to complete a workshopper strongly depends on the attendee's 
> skill set and previous experience. many people don't finish one 
> workshopper during a whole day and that is perfectly ok.
> 
> it is also depending on the skill set of the attendees which workshops 
> we /can/ offer to them. a experienced JS developer will be bored by 
> the JS fundamentals workshopper, but probably be interested in 
> something like the functional programming workshopper. I would expect 
> a mixed group of people (some with no experience at all and some 
> experienced). we should not demand that everyone has to do the JS 
> fundamentals workshop, when it would bore them.
> 
> a person who has never used javascript will probably be a bit 
> overwhelmed in the beginning and need some more help to get started. i 
> would not expect them to finish the two workshoppers and then start 
> the functional workshopper right after it.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/nodeschool/havana/issues/5#issuecomment-180440262.
> 
> Ok

## 

Grupo de Usuarios de Tecnologías Libres - Cuba
http://gutl.jovenclub.cu/

**Comment by aurium** [comment]
Hi folks! Let's detail the schedule?
When agreed we must convert this into a file.
- 09:00 – 09:50 Breakfast / Opening Session
  - Who (group) will be opening? How much time to each one?
- 10:00 – 10:50 Keynote
  - We have a big JS name at the Cubaconf?
  - I propose the theme to be _"JS to Node.js history and Node.js status"_
- 11:00 – 12:45 WorkShop / Sprints
  - I believe that should be _"JS fundamentals"_ (ONLY) and that attendants who masters that could help to fasten the neophytes learning. That will give us a more uniform group.
- 12:45 – 14:00 Lunch
- 14:00 – 17:45 three WorkShop / Sprints
  - 14:00 – 15:05 WorkShop / Sprints
    - _"Node.js fundamentals"_ and [nodesclass themes]
  - 15:20 – 16:25 WorkShop / Sprints
    - [nodesclass themes]
  - 16:40 – 17:45 WorkShop / Sprints
    - [nodesclass themes]
- 17:45 – 18:30 Closing Session
  - Who (the teachers may be nice) will be closing? How much time to each one?

## [nodesclass themes]

We should define a list of WorkShop / Sprints themes. If you agree, we can discuss this in a new thread and conclude with a file.
I'm thinking on big areas like "modularize", "Async", "HTTP", "TDD", so the attendants can focus on they interests.

**Comment by olemis** [comment]
Hi @aurium ! IMHO this ticket is perfect to discuss workshop / sprint themes .

At some point we have concluded that, considering time, it'd be nice to focus on no more than three workshops . So far we've been thinking of offering [learyounode](https://www.github.com/workshopper/learnyounode) , [javascripting](https://www.github.com/sethvincent/javascripting) and the [functional js](https://www.github.com/timoxley/functional-javascript-workshop) workshops .

I wish we could offer all them NodeSchool workshops but , for a matter of sticking to schedule , we should limit our efforts to three (that's a fact) . The first two mentioned above should be carved in stone considering that it's the first edition in Havana and these are introductory subjects . I proposed the third one because by that date David Mertz (chair of the [PSF Scientific Python WG](https://wiki.python.org/psf/ScientificWG/Charter) ) will be talking at another similar event about _Functional Programming in Python_ . Hence there was a connection between the junior-level NodeSchool courses , the advanced-level functional programming workshop , the aforementioned talk and the [Brython sprint](https://github.com/nodeschool/havana/issues/9) we plan to run on April 27th 2016 to "fill the gap" between them all .

p.s. I do not know about big JS names attending CubaConf ; nonetheless AFAIK ( @elMor3no CMIIW ) that's not a requisite to keynote . Initially I thought this was the case , but it seems the goals are different in this particular case . I'm hoping @elMor3no can tell us more about this .

**Comment by aurium** [comment]
@olemis I see. It is ok to me, but...

We have 4 WorkShop scheduled times. Starting with "JS fundamentals" (the [javascripting](https://github.com/sethvincent/javascripting)) at morning still a good idea for you?

And about the other 3 afternoon WorkShops? We will have N 1:5 groups with the same theme at the same time; or we'll mix the themes on each time, splitting by group? I see benefits of mixing, however i vote on the first option. So... I believe HTTP is a very important theme and we can do a nice and simple introduction. So, we can get 3 themes for 3 WorkShops at afternoon.

As i'm new in this group, sorry if this question is boring. I'm ok with your decision about this proposal. _(whichever it'll be)_

**Comment by olemis** [comment]
@aurium we are all new in this group and I'm under the impression that , no matter the seniority , we all should be at the same level when it comes to this kind of decisions . Ideally attendees should be able to tell us they preferences via a web form or something .

In my previous message I was just explaining why I suggested running the functional workshop to @robertkowalski [once upon a time](https://github.com/nodeschool/havana/issues/5#issuecomment-177342805) , and I know a few people interested on that subject . We can do different than that if attendees so prefer .

Nonetheless, in any case, I see `javascript` + `learnyounode` combo coming first , and after that it'd be nice to have one or two advanced subjects . My question is , aren't HTTP and Async part of `learnyounode` workshop ? Another targets I know people might be interested in are [learnyoureact](https://github.com/tako-black/learnyoureact) and [Thinking in React](https://github.com/asbjornenge/thinking-in-react) workshops. Something about Angular 1 or 2 would be very nice to have . Unless we can prepare everything to work offline , it seems to me that [npm workshop](https://github.com/npm/how-to-npm) would be difficult because [working offline is a quite probable situation](https://github.com/nodeschool/havana/issues/5#issuecomment-176991745) .

Finally I'm not sure about running [git-it](https://www.github.com/jlord/git-it) . I do not know of how proficient with git attendees are meant to be (... but again we could have a preliminary idea by setting up a web form) . What d'u think @elMor3no ?

**Comment by olemis** [comment]
Maybe we could run in parallel a special session of `git-it` for n00bs ? If so, we'd need more mentors !

**Comment by phinfonet** [comment]
I can try to create a local server using my raspberry pi and create a local connection using a router or anything like that =D

**Comment by olemis** [comment]
@aurium ... which means that `javascripting` in the morning looks fine . I am not sure about `same theme` vs `mixing themes + splitting` .

**Comment by elMor3no** [comment]
El 22/02/16 11:08, Aurélio A. Heckert escribió:

> @olemis https://github.com/olemis I see. It is ok to me, but...
> 
> We have 4 WorkShop scheduled times. Starting with "JS fundamentals" 
> (the javascripting https://github.com/sethvincent/javascripting) at 
> morning still a good idea for you?
> 
> And about the other 3 afternoon WorkShops? We will have N 1:5 groups 
> with the same theme at the same time; or we'll mix the themes on each 
> time, splitting by group? I see benefits of mixing, however i vote on 
> the first option. So... I believe HTTP is a very important theme and 
> we can do a nice and simple introduction. So, we can get 3 themes for 
> 3 WorkShops at afternoon.
> 
> As i'm new in this group, sorry if this question is boring. I'm ok 
> with your decision about this proposal. /(whichever it'll be)/
> 
> Ok...

The schedule for the 3rd day will have diferents activities: Workshops, 
Sprints, small talks, meetings...

The NodeJS WorksShop is one of diferent activities at the same time. We 
will have one rooms during all day after the Keynote.

**Comment by robertkowalski** [comment]
I really like the idea to use the morning in order to teach JavaScript fundamentals to people who are completely new to JavaScript. We did this at the first Nodeschool in Hamburg. Everyone who already knew JS would then join us after lunch.

The Nodeschool event in Hamburg would let people pick the workshoppers they like. We had one big group, it had all the folks new to JS and new to Node. The workshoppers the people in the group were working on were naturally the "JS fundamentals" and the "learnyounode" workshopper. This group required the most help from the mentors which were hovering around the room.

We additionally had several smaller groups, these folks were already using Node and wanted to try something like the Webservers or Functional Programming workshopper. The groups did not need much help from the mentors and the members of the group were able to help each other in case of a question. And sometimes we weren't able to help them, because the workshoppers they were working on were too special and we also did not know the exact answer.
